### PR TITLE
UserService Microservice 

### PR DESCRIPTION
**Pull request by venuslimm**
This PR
- Microservice UserService
- Remove unnecessary packages from Auth
- Update README.md
- Update env var

TODO:
@anthonynmh Microservice question service (to also remove unnecessary packages from QuestionService, update README.md and package.json and rename routers, controllers, etc to router, controller)
@lhy-hoyin Containerize backend services when done

### CODE REVIEW

File: User/server/route/user.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 const express = require('express');
 const router = express.Router();
 
-const userController = require('../../../backend/server/controllers/userController.js');
-const middleware = require('../middleware.js');
+const userController = require('../../../User/server/controller/userController.js');
+const middleware = require('../../../User/server/middleware.js');
```
Comment (lhy-hoyin) [inline User/server/route/user.js:5]: I propose using absolute imports (along with with relative imports). This help help to reduce the `../` that we see.

Reading on [JS import hell](https://medium.com/thefork/escaping-relative-import-paths-hell-in-javascript-9f258baeb15e) (Relative vs absolute imports).

I have already used something like this in my [UPM](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g32/blob/assignment-4-lhy/upm/jsconfig.json).

Comment (lhy-hoyin) [reply User/server/route/user.js:5]: Other parts of the code can also benefit from using absolute imports, to name a few `userService.js` `userRepoMySql.js` `AuthApi.js`

**Review by lhy-hoyin — COMMENTED** [review]
Currently, I see that the structure of `Auth` Service is like
```
Auth
-- AuthService.js
-- package*.json
-- server
-- -- controller
-- -- -- authController.js
-- -- helper
-- -- -- jwt.js
-- -- route
-- -- -- auth.js
```

I propose to simply it by flattening it to something like
```
Auth
-- AuthService.js
-- package*.json
-- AuthController.js
-- AuthRoute.js
-- helper
-- -- jwt.js
(helper can also if flattened too, if desirable)
```

File: User/server/route/user.js (hunk 1 above)
Comment (venuslimm) [reply User/server/route/user.js:5]: Resolved in #149

**Pull request merged**